txstratum/middleware.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Hathor Labs and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import re
+from typing import Awaitable, Callable, Optional
+
+from aiohttp import web
+from aiohttp.typedefs import Handler
+
+from txstratum.utils import is_version_gte
+
+Middleware = Callable[[web.Request, Handler], Awaitable[web.StreamResponse]]
+
+
+def create_middleware_version_check(
+    min_wallet_desktop_version: Optional[str],
+    min_wallet_mobile_version: Optional[str],
+    min_wallet_headless_version: Optional[str],
+) -> Middleware:
+    """Middleware factory."""
+
+    @web.middleware
+    async def version_check(
+        request: web.Request, handler: Handler
+    ) -> web.StreamResponse:
+        """Check wallet versions from user agent."""
+        user_agent = request.headers.get("User-Agent")
+
+        if not user_agent:
+            response = await handler(request)
+            return response
+
+        if min_wallet_desktop_version:
+            # Search user agent for wallet desktop string and get version
+            # Then check if version is the minimum allowed
+            # Example of a wallet desktop user agent
+            # Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) HathorWallet/0.22.1
+            # Chrome/73.0.3683.121 Electron/5.0.13 Safari/537.36 HathorWallet/0.22.1
+            wallet_desktop_regex = r"HathorWallet/(\d+\.\d+.\d+)"
+            if is_version_invalid(
+                wallet_desktop_regex, user_agent, min_wallet_desktop_version
+            ):
+                return web.json_response(
+                    {"error": "wallet-version-invalid"}, status=400
```

Comment:
Maybe `"wallet-version-too-old"`?

Reply:
I agree it's better. Done here https://github.com/HathorNetwork/tx-mining-service/pull/81/commits/61fb9ca35fdba7a7f03c4c3511c2545688fa1e38